### PR TITLE
UCS: HW timer detection bugfix

### PR DESCRIPTION
**Pull request by shamisp**
config.h defines HAVE_HW_TIMER

Signed-off-by: Pavel Shamis (Pasha) <pasharesearch@gmail.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2880/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/349/ for details.

**Comment by yosefe** [comment]
looks like this PR unveiled a bug in includes - we should avoid including ucs/arch from public headers
furthermore, we should avoid installing any ucs headers which include "config.h"
```
23:21:56 In file included from /scrap/jenkins/workspace/gh-ucx-pr__NEW/label/ARM_test_node/build-test/ucx-1.0.2168/_inst/include/ucs/arch/cpu.h:51:0,
23:21:56                  from /scrap/jenkins/workspace/gh-ucx-pr__NEW/label/ARM_test_node/build-test/ucx-1.0.2168/_inst/include/ucs/datastruct/callbackq.h:10,
23:21:56                  from /scrap/jenkins/workspace/gh-ucx-pr__NEW/label/ARM_test_node/build-test/ucx-1.0.2168/_inst/include/uct/api/uct.h:17,
23:21:56                  from /scrap/jenkins/workspace/gh-ucx-pr__NEW/label/ARM_test_node/build-test/ucx-1.0.2168/_inst/share/ucx/examples/uct_hello_world.c:9:
23:21:56 /scrap/jenkins/workspace/gh-ucx-pr__NEW/label/ARM_test_node/build-test/ucx-1.0.2168/_inst/include/ucs/arch/aarch64/cpu.h:11:20: fatal error: config.h: No such file or directory
```

File: src/ucs/arch/aarch64/cpu.h [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #ifndef UCS_AARCH64_CPU_H_
 #define UCS_AARCH64_CPU_H_
 
+#include "config.h"
```
Comment (yosefe) [inline src/ucs/arch/aarch64/cpu.h]: usually it should be wrapped with #ifdef HAVE_CONFIG_H

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2285/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
hmm. I don't think  #ifdef HAVE_CONFIG_H is a right solution here. config.h is always present.
Yes, we should avoid installation if config.h. Not sure how to do it in this case.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/355/ for details.

**Comment by shamisp** [comment]
@yosefe I updated the patch based on your suggestion

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/648/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3093/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3094/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/649/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2605/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2604/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/650/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3101/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3102/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/651/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/652/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/653/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/660/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/661/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3106/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3115/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3121/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3167/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/699/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3168/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2645/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3170/ for details.

**Comment by shamisp** [comment]
@yosefe please take a look

**Comment by yosefe** [comment]
bot:bgateNEW:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/702/ for details.

**Pull request merged**